class-events-manager-securesubmit.php

```diff
@@ -525,48 +534,40 @@ function authorize_and_capture($EM_Booking){
         global $EM_Notices;
 
         if( !class_exists('HpsServicesConfig') ){
-            require_once('hps/Hps.php');
+            require_once('hps/vendor/autoload.php');
         }
 
         $amount = $EM_Booking->get_price(false, false, true);
 
-        $config = new HpsServicesConfig();
+        $creditService = $this->setConfig();
 
-        $config->secretApiKey = get_option('em_'.$this->gateway.'_secret_key');
-        $config->versionNumber = '1740';
-        $config->developerId = '002914';
-
-        $creditService = new HpsCreditService($config);
-
-        $hpsaddress = new HpsAddress();
-        if( EM_Gateways::get_customer_field('address', $EM_Booking) != '' ) $hpsaddress->address = EM_Gateways::get_customer_field('address', $EM_Booking);
+        $hpsaddress = new Address();
+        if( EM_Gateways::get_customer_field('address', $EM_Booking) != '' ) $hpsaddress->streetAddress1 = EM_Gateways::get_customer_field('address', $EM_Booking);
         if( EM_Gateways::get_customer_field('city', $EM_Booking) != '' ) $hpsaddress->city = EM_Gateways::get_customer_field('city', $EM_Booking);
         if( EM_Gateways::get_customer_field('state', $EM_Booking) != '' ) $hpsaddress->state = EM_Gateways::get_customer_field('state', $EM_Booking);
-        if( EM_Gateways::get_customer_field('zip', $EM_Booking) != '' ) $hpsaddress->zip = EM_Gateways::get_customer_field('zip', $EM_Booking);
+        if( EM_Gateways::get_customer_field('zip', $EM_Booking) != '' ) $hpsaddress->postalCode = EM_Gateways::get_customer_field('zip', $EM_Booking);
         if( EM_Gateways::get_customer_field('country', $EM_Booking) != '' ){
             $countries = em_get_countries();
             $hpsaddress->country = $countries[EM_Gateways::get_customer_field('country', $EM_Booking)];
         }
 
         $names = explode(' ', $EM_Booking->get_person()->get_name());
-
-        $cardHolder = new HpsCardHolder();
-        if( !empty($names[0]) ) $cardHolder->firstName = array_shift($names);
-        if( implode(' ',$names) != '' ) $cardHolder->lastName = implode(' ',$names);
-        if( EM_Gateways::get_customer_field('phone', $EM_Booking) != '' ) $cardHolder->phone = EM_Gateways::get_customer_field('phone', $EM_Booking);
-        $cardHolder->email = $EM_Booking->get_person()->user_email;
-        $cardHolder->address = $hpsaddress;
-
-        $hpstoken = new HpsTokenData();
-        $hpstoken->tokenValue = $_REQUEST['securesubmit_token'];
-
-        $details = new HpsTransactionDetails();
-        $details->invoiceNumber = $EM_Booking->booking_id;
-        $details->memo = preg_replace('/[^a-zA-Z0-9\s]/i', "", $EM_Booking->get_event()->event_name);
+        
+        $hpstoken = new CreditCardData();
+        $hpstoken->token = $_REQUEST['securesubmit_token'];
```

Comment:
Let's take the opportunity to improve this:

1. Use `$_POST` if possible.
2. Test for the `securesubmit_token` index before using it.

Reply:
updated